### ISSUE TITLE
Android editor freezes on a Vulkan mobile project

### ISSUE THREAD

**Issue by m4gr3d** [godotengine/godot#88423]
### Tested versions

Reproducible in `master` build of the Android editor: v4.3.dev.custom_build [8ff821670]

### System information

Tested on a Samsung Tab S8 running Android 14

### Issue description

Opening the attached MRP project causes the Android editor to freeze with the following error visible in logcat (keeps continuously printing):

```
<gsl_ldd_control:553>: ioctl fd 89 code 0xc040094a (IOCTL_KGSL_GPU_COMMAND) failed: errno 35 Resource deadlock would occur
16:36:54.657 Adreno-GSL            W  <log_gpu_snapshot:461>: panel.gpuSnapshotPath is not set.not generating user snapshot
16:36:54.657 godot                 E  USER ERROR: Condition "err != VK_SUCCESS" is true. Returning: FAILED
16:36:54.658 godot                 E     at: command_queue_execute (drivers\vulkan\rendering_device_driver_vulkan.cpp:2188)
16:36:54.658 godot                 E  USER ERROR: Condition "err != VK_SUCCESS" is true. Returning: FAILED
16:36:54.658 godot                 E     at: fence_wait (drivers\vulkan\rendering_device_driver_vulkan.cpp:2014)
```

### Steps to reproduce

- Open the attached MRP project in the Android editor
- If that doesn't cause a freeze, run the project and navigate through it using the instructions on the screen; this usually causes a freeze as well.

### Minimal reproduction project (MRP)

[Terrain3D_demo2.zip](https://github.com/godotengine/godot/files/14317169/Terrain3D_demo2.zip)

**Comment by m4gr3d** [comment]
@DarioSamo Here's a screen recording of the issue.

https://github.com/godotengine/godot/assets/914968/e98fd339-7567-4088-a961-5f45ca8ada24

**Comment by DarioSamo** [comment]
This seems to rely on a Terrain3D add-on right? Is there any possibility the add-on relies on RD code that hasn't been updated yet?

I realize this has been marked as a regression, but on which version did this run properly exactly?

**Comment by DarioSamo** [comment]
Unable to reproduce.

I get a lot of errors in output when using the MRP as is on an Android Editor build (Samsung Galaxy S21 Mali-G78 Android 13):
```
USER SCRIPT ERROR: Parse Error: Could not find type "Terrain3D" in the current scope.
2024-02-19 11:08:12.454 10128-10169 godot                   org.godotengine.editor.v4.debug      E     at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:11)
2024-02-19 11:08:12.454 10128-10169 godot                   org.godotengine.editor.v4.debug      E  USER SCRIPT ERROR: Parse Error: Could not find type "Terrain3DEditor" in the current scope.
2024-02-19 11:08:12.454 10128-10169 godot                   org.godotengine.editor.v4.debug      E     at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:16)
2024-02-19 11:08:12.454 10128-10169 godot                   org.godotengine.editor.v4.debug      E  USER SCRIPT ERROR: Parse Error: Identifier "Terrain3DEditor" not declared in the current scope.
2024-02-19 11:08:12.454 10128-10169 godot                   org.godotengine.editor.v4.debug      E     at: GDScript::reload (res://addons/terrain_3d/editor/editor.gd:27)
```

I'm able to navigate around the environment and launch it, although I imagine the tiled textures are not intended as they don't match the OP's visuals at all. Am I missing something in regards to what setup I must do to launch this project?

**Comment by DarioSamo** [comment]
I tested with the APK build of 4.2.1 from the Godot website and tried loading the project both with the included binary of the add-on and the 0.9.1-beta of Terrain3D. Neither seem to load the project successfully even after repeated attempts, as I get an error for "Cannot get class Terrain3DSurface".

Running around the incomplete scene hasn't triggered any error for me in `master` so it seems I'm at a dead end when it comes to debugging this further.

**Comment by m4gr3d** [comment]
Closing the issue as I'm no longer able to reproduce it.
